src/box/lua/config/init.lua

```diff
@@ -420,13 +430,22 @@ function methods._print_env_list(self)
     io.stdout:write(env_source:_env_list())
 end
 
-function methods.get(self, path)
+-- Note: There is a difference between missing opts.instance and
+-- opts.instance that is equal to the given instance name. The
+-- former returns an instance configuration taking into account
+-- instance configuration sources (environment variables). The
+-- latter takes into account only cluster configuration, so the
+-- environment variables are ignored.
```

Comment:
@locker Please, let me know, whether this nuance fits good into your scenarios.

I feel it as a natural behavior, but it is possible to compare the given instance name against the name of the running instance and return the same value as if `opts.instance` is omitted.

Reply:
AFAIU, when specifying `opts.instance` different from self, we can't possibly get env. Making this behavior (ignoring env) for every value of `opts.instance` (including self) looks sane. Including env if `opts.instance` is omitted is probably fine although having a separate explicit option for that could be more straightforward.

Reply:
Thank you for the feedback!

I've filed #9824 about the explicit option that controls, whether to take env into account. I'm not sure, how the API should look, so I want to discuss/postpone it.